GameServer/Logic/Content/ItemCollection.cs

```diff
@@ -19,56 +19,43 @@ public class ContentCollectionEntry
         public string Name { get { return Convert.ToString(MetaData["Name"]); } }
         public object Id { get { return MetaData["Id"]; } }
 
-        public T GetValue<T>(string section, string name)
+        public T GetValue<T>(string section, string name, T defaultValue)
         {
-            return (T)Convert.ChangeType(Values[section][name], typeof(T));
+            try
+            {
+                return (T)Convert.ChangeType(Values[section][name], typeof(T));
+            }
+            catch (Exception)
+            {
+                return defaultValue;
+            }
         }
 
         public T SafeGetValue<T>(string section, string name, T defaultValue)
         {
             if (!Values.ContainsKey(section)) return defaultValue;
             if (!Values[section].ContainsKey(name)) return defaultValue;
-            return GetValue<T>(section, name);
-        }
-
-        public float SafeGetFloat(string section, string name, float defaultValue)
-        {
-            return SafeGetValue(section, name, defaultValue);
+            return GetValue<T>(section, name, defaultValue);
         }
 
-        public float SafeGetFloat(string section, string name)
+        public float SafeGetFloat(string section, string name, float defaultValue = 0f)
         {
-            return SafeGetFloat(section, name, 0f);
-        }
-
-        public int SafeGetInt(string section, string name, int defaultValue)
-        {
-            return SafeGetValue(section, name, defaultValue);
-        }
-
-        public int SafeGetInt(string section, string name)
-        {
-            return SafeGetInt(section, name, 0);
-        }
-
-        public string SafeGetString(string section, string name, string defaultValue)
-        {
-            return SafeGetValue(section, name, defaultValue);
+            return float.Parse(SafeGetValue(section, name, defaultValue).ToString());
         }
 
-        public string SafeGetString(string section, string name)
+        public int SafeGetInt(string section, string name, int defaultValue = 0)
         {
-            return SafeGetString(section, name, "");
+            return int.Parse(SafeGetValue(section, name, defaultValue).ToString());
```

Comment:
I don't understand the logic here; You get an `int`, cast it to `string`, and cast it to `int`. Is this necessary?